kong/plugins/key-auth/handler.lua

```diff
@@ -179,9 +237,44 @@ local function do_authentication(conf)
     return nil, { status = 500, message = "An unexpected error occurred" }
   end
 
-  set_consumer(consumer, credential)
+  if conf.validate_signature then
+    -- 4. Verify sha256
+    local secret = credential.secret
+    local now = math.floor(ngx.time())
+    local sha = sha256:new()
+    local ks = key .. secret
+
+
+    local verify_sig = function(time)
+      sha:reset()
+      sha:update(ks)
+      sha:update(tostring(time))
+      local final = to_hex(sha:final())
+      --print("  verifying - sha(" .. ks .. " .. " .. tostring(time) .. ") = " .. final)
+      -- print(signature .. " == " .. final)
+      return signature == final
+    end
```

Comment:
Closures don't play well with the JIT and this may negatively (and _significantly_) impact the performance of this plugin. Please find a way to write it in a more JIT-friendly way (avoiding closures, that is).